### ISSUE TITLE
Copies chat colors even when settings are false

### ISSUE THREAD

**Issue by Fluboxer** [tierrif/chat-utils#29]
**Describe the bug**
Title. It just ignores setting, giving you &b colors and &lstuff

**To Reproduce**
1. Copy message involving chat colors 
2. thats it 

**Expected behavior**
Not this

**Screenshots**
while screenshot for #28 is suitable as you can see wrong colors, I have more... noticeable example
![2023-07-19_17 50 28](https://github.com/tierrif/chat-utils/assets/36457056/c3f884f4-cc27-4079-b29e-9a5d76b98cdc)

**Client information**
 - Minecraft: Fabric 1.20.1 
 - Mod version: **1.10.0+1.20.1**
 - Other mods: 
 ![Screenshot_20230719_171807](https://github.com/tierrif/chat-utils/assets/36457056/4ebf7ca9-e7a3-4a7f-989c-1855599f5d5c)
- This time I also have settings to show: (also used in #28 )
![2023-07-19_17 46 19](https://github.com/tierrif/chat-utils/assets/36457056/80bdd692-dd77-4e85-baae-2e53af76b52b)

**Additional context**
Obviously, just like previous issue, that one is new version only. Previous mod version is just fine

**Comment by tierrif** [comment]
When it comes to the setting, it's inverted, temporarily "enable" color copying and it'll stop copying color codes. For the missing '&' symbols now and then, it's apparently because the § symbol isn't being fully replaced. I will look into this.

**Comment by Fluboxer** [comment]
Yes, it seems like copy colors is inverted. It also don't eat all color codes 

Here is OFF and ON on the setting (on is second one, but some color codes remain)
Message indeed looks like this: "[13:26:21] [Render thread/INFO]: [CHAT] You already ate this cake recently! §8[§cx3§8]"

![изображение](https://github.com/tierrif/chat-utils/assets/36457056/bf6c17d9-9edb-436e-ab3a-dfcb50d3bdf0)
![изображение](https://github.com/tierrif/chat-utils/assets/36457056/a1645c87-6d32-4546-88a2-deb196e5371c)

**Comment by Fluboxer** [comment]
It's still a thing 💀 
fix fixed most of the cases (for example, I can properly copy all stuff above) - but there are some rare occurrences when it still happens 
in case of Hypixel Skyblock it happens when you attempt to copy auction bought message. Both copy colors options are off

![изображение](https://github.com/tierrif/chat-utils/assets/36457056/3002a166-b687-49ce-86a5-b21cef016415)

**Comment by retrixe** [comment]
https://github.com/tierrif/chat-utils/issues/35#issuecomment-1677122672 this comment of yours is the same issue as what you are describing here

I'm aware of the cause and will look into fixing it in my spare time, or @tierrif will if he feels like it, I talked to him about the underlying cause just now as well. I will reopen this issue and ignore that comment for now (the other issue described in it is likely #28).

<details>
<summary>Chat logs for my own (or anyone else's) ref</summary>

[04:33] when you switch from Minecraft Text to Adventure
[04:33] Text supports having colour codes in {"content":"&f over here with section sign instead of & obv"}
[04:34] Adventure doesn't because that's stupid
[04:34] so when you cross over then it just becomes <section sign><letter>
[04:34] I wrote a workaround but ig it's imperfect
[04:34] so we will need to bring back stripColor and translateAlternateColorCodes after running through PlainTextComponentSerialiser
</details>
